cms/djangoapps/course_creators/admin.py

```diff
@@ -30,26 +33,53 @@ def get_email(obj):
 get_email.short_description = 'email'
 
 
+class CourseCreatorForm(forms.ModelForm):
+    """
+    Admin form for course creator
+    """
+    class Meta:
+        model = CourseCreator
+        fields = '__all__'
+
+    def clean(self):
+        """
+        Running validation of the form. Validation to stop user from entering
+        false data.
+        """
+        all_orgs = self.cleaned_data.get("all_organizations")
+        updated_state = self.cleaned_data.get("state")
+        orgs = self.cleaned_data.get("orgs").exists()
+        add_role = updated_state == CourseCreator.GRANTED
+        if not add_role and orgs:
+            raise ValidationError("Organizations cannot be added if the role is not granted.")
+        if orgs and all_orgs:
+            raise ValidationError("All Organization should be disabled to add organization")
+        if not orgs and not all_orgs:
+            raise ValidationError("Organizations should be added if All Organization is disabled.")
```

Comment:
@farhaanbukhsh, maybe these two could be combined into a single `if` condition and a combined error message like.

``The role can be granted either to all organizations or to specific organizations but not both.```
?